src/utils/types.js

```diff
@@ -0,0 +1,10 @@
+const type = {
+  isArray: (value) => value && typeof value === 'object' && value.constructor === Array,
+  isBoolean: (value) => typeof value === 'boolean',
+  isFunction: (value) => typeof value === 'function',
+  isNumber: (value) => typeof value === 'number' && Number.isFinite(value),
```

Comment:
This might be useful, but do you expect to see many infinite numbers here?

Reply:
It's to handle `NaN`:

```js
const checks1 = [0, -1, 1, 3.333, NaN, -NaN].map((value) => {
	return typeof value === 'number' && Number.isFinite(value);
});

const checks2 = [0, -1, 1, 3.333, NaN, -NaN].map((value) => {
	return typeof value === 'number';
});

console.log(checks1); 
console.log(checks2);
```

Output:

```
Array(6) [ true, true, true, true, false, false ]
Array(6) [ true, true, true, true, true, true ]
```

Reply:
I was just trying to explain that for most cases you really only need to use `typeof`, but you use these as part of your validation framework so I retract my point for this case.